crates/re_types/src/blueprint/components/zoom_behavior_ext.rs

```diff
@@ -0,0 +1,25 @@
+use super::ZoomBehavior;
+
+// TODO(#3384): Code-gen all of this
+// Pseudo-enum
+#[allow(non_upper_case_globals)]
+impl ZoomBehavior {
+    pub const PreserveAspectRatio: ZoomBehavior = ZoomBehavior(1);
+    pub const LockToRange: ZoomBehavior = ZoomBehavior(2);
+}
+
+impl Default for ZoomBehavior {
+    fn default() -> Self {
+        Self::PreserveAspectRatio
+    }
+}
+
+impl std::fmt::Display for ZoomBehavior {
```

Comment:
we're a bit inconsistent with which of these enum components/datatypes get an `as_str` method and which one get a `Display` fmt. I suppose whenever we can have a `const fn as_str` we should have both (I previously in fact removed an `as_str` because I wanted the `Display` one but wasn't entirely the right move I think now). Notably `MarkerShape` has only `as_str`

Reply:
Ended up getting rid of the enum for now.